manifest: openthread: Regular openthread upmerge to `6d55738`

canisLupus1313 opened the pull request:
Openthread upmerge to `6d55738`. Includes implementing of:
- `OPENTHREAD_CONFIG_RADIO_STATS_ENABLE`
- `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE`

Commit:
manifest: openthread: Regular openthread upmerge to `6d55738`

Openthread upmerge to `6d55738`. Includes implementing of:
- `OPENTHREAD_CONFIG_RADIO_STATS_ENABLE`
- `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE`

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

pdgendt commented on modules/openthread/Kconfig.features:
```diff
@@ -51,6 +51,9 @@ config OPENTHREAD_BORDER_ROUTING
 	bool "Border routing counters support"
 
+config OPENTHREAD_BORDER_ROUTING_DHCP6_PD_ENABLE
+	bool "Dhcpv6 pd support in border routing"
```
Nit:
```suggestion
	bool "DHCPv6 pd support in border routing"
```

pdgendt commented on modules/openthread/platform/openthread-core-zephyr-config.h:
```diff
@@ -446,4 +446,15 @@
+ * Enable support for Radio Statistics.
+ *
+ */
+#ifndef CONFIG_OPENTHREAD_RADIO_STATS_ENABLE
+#define OPENTHREAD_CONFIG_RADIO_STATS_ENABLE CONFIG_OPENTHREAD_CONFIG_RADIO_STATS_ENABLE
+#endif
```
Inverse logic and copy-paste issue?

```suggestion
#ifdef CONFIG_OPENTHREAD_RADIO_STATS_ENABLE
#define OPENTHREAD_CONFIG_RADIO_STATS_ENABLE CONFIG_OPENTHREAD_RADIO_STATS_ENABLE
#endif
```

pdgendt reviewed (commented):
Some compliance checks still need to be addressed.

edmont commented on modules/openthread/Kconfig.features:
```diff
@@ -51,6 +51,9 @@ config OPENTHREAD_BORDER_ROUTING
 	bool "Border routing counters support"
 
+config OPENTHREAD_BORDER_ROUTING_DHCP6_PD_ENABLE
+	bool "DHCPv6 pd support in border routing"
```
```suggestion
	bool "DHCPv6-PD support in border routing"
```

pdgendt commented on modules/openthread/platform/openthread-core-zephyr-config.h:
```diff
@@ -446,4 +446,15 @@
 #endif
 
+
+ /**
+ * @def OPENTHREAD_CONFIG_RADIO_STATS
```
Nit:
```suggestion
/**
 * @def OPENTHREAD_CONFIG_RADIO_STATS
```

rlubos commented:
Seems that Babble sim tests have some trouble running. The devices start to communicate eventually, but there is some disturbance early in the test, due to both devices forming own mesh early (i. e. starting as `Leader`). One of them degrades to a `Child` after a while and joins the other mesh.

I'm not really sure why would it start to manifest on this PR though. @aescolar I'm curious, would it be possible to delay the execution of the second device for a few seconds, so that the first one has a chance to form a mesh before the other starts? Ideally, we could start the server device, wait a few seconds, and then start the client, which would then join the already existing mesh, instead of forming its own.

aescolar commented:
> ... @aescolar I'm curious, would it be possible to delay the execution of the second device for a few seconds, so that the first one has a chance to form a mesh before the other starts? Ideally, we could start the server device, wait a few seconds, and then start the client, which would then join the already existing mesh, instead of forming its own.

Yes, certainly, just add  `-start_offset=2e6`  to the second device command line to delay it by 2 seconds (2 million µs) of simulated time

pdgendt commented on tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh:
```diff
@@ -18,7 +18,7 @@ EXECUTE_TIMEOUT=100
 
 Execute ./bs_${BOARD}_tests_bsim_net_sockets_echo_test_prj_conf_overlay-802154_conf \
-  -v=${verbosity_level} -s=${simulation_id} -d=0 -RealEncryption=1 \
+  -v=${verbosity_level} -s=${simulation_id} -start_offset=2e6 -d=0 -RealEncryption=1 \
```
Create a separate commit for this change, please.

Commit:
bsim: net: Add delay to echo client in ot bsim tests.

This change ensures that openthread network topology will be formed
properly, by delaing start of echo client application.

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

rlubos commented:
@aescolar Any clue why are the Babblesim tests for OT failing now? The packet exchange rate seems decent now, but the test fails with:
> d_00: @00:00:23.998575  main: The TESTCASE FAILED with return code 1

aescolar commented:
Due to the 2s delay, the testcase just did not have time to pass.
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/net/sockets/echo_test/src/echo_test.c#L18
(There was only 1 second margin before).
You can just increase the maximum simulation duration by 5 seconds:

```diff
diff --git a/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh b/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh
index 3516426122..a5b1f856d4 100755
--- a/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh
+++ b/tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh
@@ -25,6 +25,6 @@ Execute ./bs_${BOARD}_samples_net_sockets_echo_server_prj_conf_overlay-ot_conf\
   -v=${verbosity_level} -s=${simulation_id} -d=1 -RealEncryption=1 \
 
 Execute ./bs_2G4_phy_v1 -v=${verbosity_level} -s=${simulation_id} \
-  -D=2 -sim_length=26e6 -argschannel -at=40 -argsmain $@
+  -D=2 -sim_length=30e6 -argschannel -at=40 -argsmain $@
 
 wait_for_background_jobs #Wait for all programs in background and return != 0 if any fails
```

rlubos commented:
Nice, it seems Babblesim tests are finally passing. Thanks @canisLupus1313 for addressing the problem and thanks @aescolar for your support.

rlubos commented on west.yml:
```diff
@@ -290,7 +290,7 @@ manifest:
       path: modules/lib/open-amp
     - name: openthread
-      revision: d9abe3071c0131a4adb5d7e7451319b735e6d855
+      revision: pull/97/head
```
`6d557383e8bc8654f813674e8631c2e56b8e70d0`

canisLupus1313 replied:
done
